### PR TITLE
feat(osd): extend Routes API

### LINKED ISSUE #753: Routes API enhancements
There are some important limitations in the current routes API which should probably be addressed with the v0.2.0 API:

- [ ] No support for IPv6 (routes query is IPv4-only; no facility to choose, in API)
- [ ] No route metric returned (critical to diagnosing conflicting route problems)
- [ ] No protocol returned (cannot tell how the route got in the table)
- [ ] No table ID returned (needed for systems with multiple routing tables)
- [ ] No scope returned (probably of lesser significance, but useful filter point, especially for IPv6 addresses)

After internally griping about the library that is being used, I realize that the fault is not the library, but with the kernel API itself.  Netlink is pretty painful.

Regardless, I have most of this addressed in a local branch.  I'll try to get a PR  up soon, barring any objections in scope.

### PR DESCRIPTION
**Pull request by Ulexus**
- Add support for IPv6, route metrics, protocols, sources, families.
- Fix destination string format to be CIDR.

Fixes #753

Signed-off-by: Seán C McCord <ulexus@gmail.com>

### CODE REVIEW

File: internal/app/osd/internal/reg/reg.go [hunk 1]
```diff
@@ -321,40 +322,46 @@ func (r *Registrator) Logs(req *proto.LogsRequest, l proto.OSD_LogsServer) (err
...
-			Interface:   link.Attrs().Name,
-			Destination: destination,
-			Gateway:     gateway,
+		if ifaceData.Attributes != nil {
+			ifaceName = ifaceData.Attributes.Name
 		}
-		routes = append(routes, routeMessage)
-	}
 
-	data = &proto.RoutesReply{Routes: routes}
```
Comment (bradbeam) [inline internal/app/osd/internal/reg/reg.go]: we do have a named return in the func definition, so we probably want to make use of it or drop it.

File: internal/app/osd/internal/reg/reg_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package reg
+
+import (
+	"net"
+	"testing"
+
+	"golang.org/x/sys/unix"
+)
+
+func TestToCIDR(t *testing.T) {
```
Comment (bradbeam) [inline internal/app/osd/internal/reg/reg_test.go]: nit, we've been standardizing on using testify for our tests. ex, https://github.com/talos-systems/talos/blob/master/internal/app/init/pkg/system/system_test.go

File: internal/app/osd/proto/api.proto [hunk 3]
```diff
@@ -77,9 +77,61 @@ message RoutesReply { repeated Route routes = 1; }
...
+   IPV6 = 0xa;
+}
+
+enum RouteProtocol {
+   RTPROT_UNSPEC = 0;
+   RTPROT_REDIRECT = 1;  // Route installed by ICMP redirects
+   RTPROT_KERNEL = 2;    // Route installed by kernel
+   RTPROT_BOOT = 3;      // Route installed during boot
+   RTPROT_STATIC = 4;    // Route installed by administrator
+   RTPROT_GATED = 8;	    // Route installed by gated
```
Comment (andrewrynhard) [inline internal/app/osd/proto/api.proto]: This is a super nit 😄 : Comments aren't aligned, but it looks like you meant to have them aligned.

File: internal/app/osd/internal/reg/reg.go (hunk 1 above)
Comment (Ulexus) [reply internal/app/osd/internal/reg/reg.go]: Quite right, thanks; given the previous return branches, the cleanest is just to remove the return definitions.  If this is a style problem, though, I'm happy to revert.

File: internal/app/osd/internal/reg/reg_test.go (hunk 2 above)
Comment (Ulexus) [reply internal/app/osd/internal/reg/reg_test.go]: Thanks; I should have checked.

File: internal/app/osd/proto/api.proto (hunk 3 above)
Comment (Ulexus) [reply internal/app/osd/proto/api.proto]: Thanks; looks like some tabs remained from the copy-paste.

**Pull request merged**